### PR TITLE
[DOC] Update Hive doc page with the 4.0.0-alpha-1 features

### PR DESCRIPTION
**Pull request by pvary**
Update Hive doc page with the 4.0.0-alpha-1 features

### CODE REVIEW

File: docs/hive/_index.md
Comment (szlta) [inline docs/hive/_index.md:371]: not CHANGE COLUMN, it's REPLACE COLUMNS
Also note, that dropping columns is only thing REPLACE COLUMNS can be used for i.e. if columns are specified out-of-order an error will be thrown signalling this limitation.

File: docs/hive/_index.md
Comment (pvary) [reply docs/hive/_index.md]: Done

File: docs/hive/_index.md
Comment (pvary) [reply docs/hive/_index.md]: Done

File: docs/hive/_index.md
Comment (pvary) [reply docs/hive/_index.md]: Done

File: docs/hive/_index.md
Comment (pvary) [reply docs/hive/_index.md]: Done

**Comment by pvary** [comment]
> Generally looks good, some minor remarks.

Thanks @szlta for the review!
Could you please check the changes?

Thanks,
Peter

**Review by szlta — APPROVED** [review]
Looks good, thanks for the corrections.
+1 non-binding
PS: feel free to resolve my comments - seems like I don't have permission to do so...

**Comment by lcspinter** [comment]
Thanks @pvary for your work! 
+1 non-binding

File: docs/hive/_index.md
Comment (samredai) [inline docs/hive/_index.md:29]: These versions still support this so it should be present tense, right? So `s/supported/supports`

File: docs/hive/_index.md
Comment (samredai) [inline docs/hive/_index.md:34]: Can `(will append rows)` and `(MapReduce only)` be expanded to warning/info boxes? Something along the lines of:
```
{{< hint info >}}
Inserting into a table will append rows to the relevant partitions.
{{< /hint >}}
{{< hint warning >}}
Inserting into a table works with MapReduce only.
{{< /hint >}}
```

File: docs/hive/_index.md
Comment (samredai) [inline docs/hive/_index.md:37]: nit: I'd word this as

> With Hive version 4.0.0-alpha-1 and above, the Iceberg `HiveCatalog` supports the following additional features:

File: docs/hive/_index.md
Comment (samredai) [inline docs/hive/_index.md:40]: I don't think the parentheses are needed here. I'd suggest:
```
* Creating an Iceberg table with any partition spec, including the various transforms supported by Iceberg
```

File: docs/hive/_index.md
Comment (samredai) [inline docs/hive/_index.md:49]: Similar to my other comment, I would put these in warning or info boxes instead of tacking them onto the end of the line. Those boxes are more visible disclaimers.

File: docs/hive/_index.md
Comment (samredai) [inline docs/hive/_index.md:56]:
```suggestion
Hive 4.0.0-alpha-1 comes with the Iceberg 0.13.1 included. No additional downloads or jars are needed.
```

File: docs/hive/_index.md
Comment (samredai) [inline docs/hive/_index.md:62]: This could just be one line right?

> In order to use Hive 2.3.x or Hive 3.1.x, you must load the Iceberg-Hive runtime jar and enable Iceberg support, either globally or for an individual table using a table property.

and then jump into the sections below that provide more details.

File: docs/hive/_index.md
Comment (samredai) [inline docs/hive/_index.md:93]: Keep in mind these changes may impact what shows up in the table of contents for this page. IIRC, h5 headers don't show up in the TOC.

File: docs/hive/_index.md
Comment (pvary) [reply docs/hive/_index.md]: Done

File: docs/hive/_index.md
Comment (pvary) [reply docs/hive/_index.md]: Done

File: docs/hive/_index.md
Comment (pvary) [reply docs/hive/_index.md:100]: I think this is fine.

**Comment by samredai** [comment]
@pvary I opened a [PR](https://github.com/apache/iceberg-docs/pull/98) with these changes directly against the `latest` branch and we'll need to do the same for the `0.13.2` branch. That way this can get released with the current format of the docs site and I'll make sure those changes are carried over to the new site format.

**Comment by samredai** [comment]
We'll have to wait for @rdblue or another committer to review it and merge

**Comment by pvary** [comment]
Thanks @samredai for the help. With Ryan's review, I was able to merge https://github.com/apache/iceberg-docs/pull/98

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Looks good to me. Can you rebase? I don't think that we need the _index.md file any more. @samredai, how should we update this?

**Comment by pvary** [comment]
@samredai: Any other change that we have to keep from the changes came in for docs/hive/_index.md? What I see are only formatting changes

**Comment by samredai** [comment]
Ah that's right, this has already been added by PR #5115 . That includes all of the changes from this PR here so this should be good to close.

**Comment by pvary** [comment]
> Here's a rebased PR #5161 with the content added to the existing `hive.md` file. @pvary can you confirm the content all looks good? Then we can merge that and close this PR.

Will check it tomorrow. It is quite late here, and the change is big.

Thanks for all the help!

**Comment by pvary** [comment]
Merged #5161 to master.
Thanks @samredai and @krishahn for the help!
Thanks @szlta, @lcspinter and @rdblue for the reviews!

**Pull request closed without merging**
